pkg/kubelet/dockershim/docker_service.go

```diff
@@ -148,9 +148,55 @@ type dockerNetworkHost struct {
 
 var internalLabelKeys []string = []string{containerTypeLabelKey, containerLogPathLabelKey, sandboxIDLabelKey}
 
+// ClientConfig is parameters used to initialize docker client
+type ClientConfig struct {
+	DockerEndpoint            string
+	RuntimeRequestTimeout     time.Duration
+	ImagePullProgressDeadline time.Duration
+
+	IsFakeClient bool
+
+	// Configuration for fake docker client
+	EnableSleep       bool
+	WithTraceDisabled bool
+}
+
+// NewDockerClientFromConfig create a docker client from given configure
+// return nil if nil configure is given.
+func NewDockerClientFromConfig(config *ClientConfig) libdocker.Interface {
+	if config != nil {
+		if config.IsFakeClient {
+			// Create fake docker client.
```

Comment:
This feels wrong ... (mixing code used during testing)

Reply:
I'm trying to avoid import `libdocker` in hollow_kubelet.

So, how about add a `NewFakeDockerClient(...)` in dockershim?

Reply:
OK. Seems the external use of `fakeDockerClient` is only in `hollow_nod` and Nvidia GPU test. I think it's OK for them call `libdocker.NewFakeDockerClient` as they are all for test. So I can remove this buggy logic here.